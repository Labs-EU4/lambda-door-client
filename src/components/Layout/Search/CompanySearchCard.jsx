import React from 'react';
import styled from 'styled-components';
import { Card } from 'antd';

const CompanySearchCard = ({ history, name, location, website, type, id }) => {
  return (
    <StyledCard onClick={() => history.push(`/company-page/${id}`)}>
      <h2>{name}</h2>
      <p>{location}</p>
      <a href={website}>{website}</a>
      <p>
        Type: &nbsp;
        {type}
      </p>
    </StyledCard>
  );
};

export default CompanySearchCard;

const StyledCard = styled(Card)`
  height: 200px;
  width: 350px;
  margin-right: 1rem !important;
  margin-bottom: 1rem !important;
  &:hover {
    border: 2px solid #bb1333;
    cursor: pointer;
  }
`;
