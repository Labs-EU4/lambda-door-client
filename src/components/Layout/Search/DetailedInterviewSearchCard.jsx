/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import { Button, Icon, Card } from 'antd';
import { connect } from 'react-redux';
import { useParams, withRouter } from 'react-router-dom';
import styled from 'styled-components';

const DetailedInterviewSearchCard = ({
  search: { searchResults },
  history,
}) => {
  const reviewId = useParams().id;
  const review = searchResults[0].find(elem => elem.id === Number(reviewId));

  const handleCompanyClick = e => {
    e.stopPropagation();
    history.push(`/company-page/${review.company_id}`);
  };

  return (
    <InterviewCardContainer>
      <Button
        style={{
          marginBottom: '30px',
          border: '1px solid #BB1333',
          color: '#BB1333',
        }}
        onClick={() => history.goBack()}
      >
        <Icon type="left" />
        Back to Results
      </Button>
      <div>
        <div className="title-div">
          <h1 onClick={handleCompanyClick}>{review.name}</h1>
        </div>
        <div>{review.text}</div>
        <p>
          Job Type: &nbsp;
          {review.interest}
        </p>
        <div className="bottom">
          <div className="contact">
            {review.is_accepting_questions ? (
              <p>
                Have questions? &nbsp;
                <Button>Contact Me</Button>
              </p>
            ) : (
              ''
            )}
          </div>
          <div className="username">{review.full_name}</div>
        </div>
      </div>
    </InterviewCardContainer>
  );
};

export default withRouter(connect(state => state)(DetailedInterviewSearchCard));

const InterviewCardContainer = styled.div`
  max-width: 500px;
  p {
    padding-top: 2rem;
  }
  h1 {
    margin-bottom: 0;
    padding-bottom: 35px !important;
    font-size: 20px;
    font-weight: 900;
    transition: 1s hover;
    &:hover {
      opacity: 0.6;
      cursor: pointer;
    }
    &:active {
      transform: scale(1.05);
    }
  }
`;
