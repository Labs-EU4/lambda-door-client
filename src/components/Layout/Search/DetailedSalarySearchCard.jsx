import React from 'react';
import { Button, Icon, Skeleton, Card } from 'antd';
import { connect } from 'react-redux';
import { useParams, withRouter } from 'react-router-dom';
import styled from 'styled-components';
import currencies from '../../../utils/currencies';

let salaryFormatted;

const DetailedSalarySearchCard = ({ search: { searchResults }, history }) => {
  const reviewId = useParams().id;
  const review = searchResults[0].find(elem => elem.id === Number(reviewId));

  const currencyUnit = currencies.find(curr => curr.name === review.currency)
    .symbol;
  salaryFormatted = `${currencyUnit}${Number(review.salary)
    .toFixed(2)
    .replace(/\d(?=(\d{3})+\.)/g, '$&,')}`;

  const handleCompanyClick = e => {
    e.stopPropagation();
    history.push(`/company-page/${review.company_id}`);
  };

  return (
    <SalaryCardContainer>
      <Button
        style={{
          marginBottom: '30px',
          border: '1px solid #BB1333',
          color: '#BB1333',
        }}
        onClick={() => history.goBack()}
      >
        <Icon type="left" />
        Back to Results
      </Button>
      <div className="title-div">
        <h1 onClick={handleCompanyClick}>{review.name}</h1>
      </div>

      <div className="salary-div">
        <p>
          {salaryFormatted}
          /yr
        </p>
      </div>

      <div className="interest">
        <h3>Job Category</h3>
        <p>{review.interest}</p>
      </div>

      <div className="description">
        <h3>Job Description</h3>
        <p>{review.description}</p>
      </div>

      <div className="bottom">
        <div className="contact">
          {review.is_accepting_questions ? (
            <p>
              Have questions? &nbsp;
              <Button>Contact Me</Button>
            </p>
          ) : (
            ''
          )}
        </div>
        <div className="username">{review.full_name}</div>
      </div>
    </SalaryCardContainer>
  );
};

export default withRouter(connect(state => state)(DetailedSalarySearchCard));

const SalaryCardContainer = styled.div`
  div {
    margin-bottom: 2rem;
  }

  h3 {
    font-weight: bold !important;
  }
  h1 {
    margin-bottom: 0;
    padding-top: 35px !important;
    font-size: 20px;
    font-weight: 900;
    transition: 1s hover;
    &:hover {
      opacity: 0.6;
      cursor: pointer;
    }
    &:active {
      transform: scale(1.05);
    }
  }

  .salary-div {
    font-size: 1.5rem;
  }
`;
