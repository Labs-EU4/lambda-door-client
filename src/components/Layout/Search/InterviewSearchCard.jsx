import React from 'react';
import styled from 'styled-components';
import { Card } from 'antd';

const InterviewSearchCard = ({ history, name, text, interest, id }) => {
  return (
    <StyledCard onClick={() => history.push(`/search-results/interview/${id}`)}>
      <h2>{name}</h2>
      <p>{interest}</p>
      <p>
        {text.slice(0, 100)}
        ...
      </p>
    </StyledCard>
  );
};

export default InterviewSearchCard;

const StyledCard = styled(Card)`
  height: 200px;
  width: 350px;
  height: 200px;
  width: 350px;
  margin-right: 1rem !important;
  margin-bottom: 1rem !important;
  &:hover {
    border: 1px solid #bb1333;
    cursor: pointer;
  }
`;
