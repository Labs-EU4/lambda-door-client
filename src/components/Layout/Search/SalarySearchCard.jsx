/* eslint-disable camelcase */
import React from 'react';
import styled from 'styled-components';
import { Card } from 'antd';
import currencies from '../../../utils/currencies';

let salaryFormatted;

const SalarySearchCard = ({
  history,
  name,
  salary,
  job_title,
  currency,
  id,
}) => {
  const currencyUnit = currencies.find(curr => curr.name === currency).symbol;
  salaryFormatted = `${currencyUnit}${Number(salary)
    .toFixed(2)
    .replace(/\d(?=(\d{3})+\.)/g, '$&,')}`;

  return (
    <StyledCard onClick={() => history.push(`/search-results/salary/${id}`)}>
      <h2>{name}</h2>
      <p className="salary">
        {salaryFormatted}
        /yr
      </p>
      <p>{job_title}</p>
    </StyledCard>
  );
};

export default SalarySearchCard;

const StyledCard = styled(Card)`
  height: 200px;
  width: 350px;
  margin-right: 1rem !important;
  margin-bottom: 1rem !important;
  &:hover {
    border: 2px solid #bb1333;
    cursor: pointer;
  }

  .salary {
    font-size: 1.5rem;
  }
`;
