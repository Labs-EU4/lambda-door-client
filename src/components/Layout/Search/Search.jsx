/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable no-nested-ternary */
import React, { useState } from 'react';
import { Input, Select, Form, Button, Icon } from 'antd';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { mobilePortrait, tabletPortrait } from '../../../styles/theme.styles';
import getSearchResults from '../../../state/actions/search';

const { Option } = Select;

const SearchForm = ({
  getSearchResults,
  history,
  searchVisible,
  setSearchVisible,
}) => {
  const [formValues, setFormValues] = useState({
    search_category: 'companies',
    search_query: '',
  });

  const handleChange = event => {
    setFormValues({ ...formValues, [event.target.name]: event.target.value });
  };

  const handleComponentChange = name => {
    setFormValues({
      ...formValues,
      search_category: name,
    });
  };

  const handleSubmit = e => {
    e.preventDefault();
    getSearchResults(formValues);
    setSearchVisible(false);
    history.push('/search-results');
  };

  const closeSearch = e => {
    e.preventDefault();
    setSearchVisible(false);
  };

  return (
    <StyledFormContainer searchVisible={searchVisible}>
      <Form>
        <div className="search-category">
          <Select
            name="search_category"
            defaultValue="companies"
            onChange={handleComponentChange}
            size="large"
          >
            <Option value="companies">Companies</Option>
            <Option value="salaries">Salaries</Option>
            <Option value="interviews">Interviews</Option>
          </Select>
        </div>
        <div className="search-field">
          <Input
            placeholder={
              formValues.search_category === 'companies'
                ? 'Company Name, Location'
                : formValues.search_category === 'salaries'
                ? 'Job Title, Location'
                : 'Company Name, Job Title'
            }
            size="large"
            name="search_query"
            onChange={handleChange}
          />
        </div>
        <Button onClick={handleSubmit} size="large">
          Search
        </Button>
        <CloseSearch onClick={closeSearch}>
          <Icon type="close" />
        </CloseSearch>
      </Form>
    </StyledFormContainer>
  );
};

export default withRouter(
  connect(state => state, { getSearchResults })(SearchForm)
);

const StyledFormContainer = styled.div`
  position: relative;
  .ant-form {
    max-width: 800px;
    width: 100%;
    display: flex;
    flex-direction: row;
    @media ${tabletPortrait} {
      position: fixed;
      left: 0;
      transition: all 0.25s linear;
      bottom: ${props => (props.searchVisible ? '0' : '-220px')};
      flex-direction: column;
      justify-content: center;
      align-items: center;
      width: 100vw;
      height: 180px;
      padding: 1rem 0;
    }
    @media ${mobilePortrait} {
      position: fixed;
      left: 0;
      transition: all 0.25s linear;
      bottom: ${props => (props.searchVisible ? '0' : '-220px')};
      flex-direction: column;
      justify-content: center;
      align-items: center;
      width: 100vw;
      height: 180px;
      padding: 1rem 0;
    }
    .ant-btn {
      margin-left: 1rem;
      @media ${mobilePortrait} {
        width: 90vw;
        margin-left: 0;
      }
    }
    .ant-input {
      width: 500px;
      background-color: #ffffff !important;
      @media ${tabletPortrait} {
        width: 90vw;
      }
      @media ${mobilePortrait} {
        width: 90vw;
      }
    }

    @media ${tabletPortrait} {
      display: flex;
      justify-content: space-between;
      background-color: grey;
      width: 100vw;
      z-index: 500;
    }
    @media ${mobilePortrait} {
      display: flex;
      justify-content: space-between;
      background-color: grey;
      width: 100vw;
      z-index: 500;
    }
    .search-category {
      min-width: 150px;
      margin-right: 1rem;
      @media ${tabletPortrait} {
        width: 90vw;
        margin-right: 0;
      }
      @media ${mobilePortrait} {
        width: 90vw;
        margin-right: 0;
      }
    }
    .mobile-logo-btn {
      display: none;
      @media ${tabletPortrait} {
        display: inherit;
        width: 50px;
        padding: 12px;
        img {
          width: 1.88rem;
        }
      }
      @media ${mobilePortrait} {
        display: inherit;
        width: 50px;
        padding: 12px;
        img {
          width: 1.88rem;
        }
      }
    }

    @media ${tabletPortrait} {
      background-color: #bb1333;
    }
    @media ${mobilePortrait} {
      background-color: #bb1333;
    }
  }
`;

const CloseSearch = styled.div`
  display: none;
  position: absolute;
  top: -1.5rem;
  right: 0;
  background-color: #bb1333;
  color: white;
  padding: 0.5rem;
  height: 2rem;
  width: 3rem;
  justify-content: center;
  align-items: center;
  border-radius: 1rem 0 0 0;

  @media ${tabletPortrait} {
    display: flex;
  }
  @media ${mobilePortrait} {
    display: flex;
  }
`;
