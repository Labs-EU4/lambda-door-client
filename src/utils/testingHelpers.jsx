import React from 'react';
import { createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import rootReducer from '../state/reducers/index';

export const TestingRouter = ({
  ComponentWithRedirection,
  RedirectUrl,
  MainRoute,
}) => (
  <>
    <Route
      path={MainRoute}
      exact={true}
      render={props => {
        return <ComponentWithRedirection {...props} />;
      }}
    />
    <Route path={RedirectUrl} render={() => <div>{RedirectUrl}</div>} />
  </>
);

export const renderWithRedux = (
  component,
  {
    initialState,
    store = createStore(
      rootReducer,
      initialState,
      compose(applyMiddleware(thunk))
    ),
  } = {},
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>
      </Provider>
    ),
    store,
    history,
  };
};
