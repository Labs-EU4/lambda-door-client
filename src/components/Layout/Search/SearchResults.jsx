/* eslint-disable no-nested-ternary */
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Spin, Button, Icon } from 'antd';
import styled from 'styled-components';
import CompanySearchCard from './CompanySearchCard';
import SalarySearchCard from './SalarySearchCard';
import InterviewSearchCard from './InterviewSearchCard';
import {mobilePortrait} from '../../../styles/theme.styles';

const SearchResults = ({ search: { isSearching, searchResults }, history }) => {
  console.log('is searching', isSearching);
  return (
    <Container>
      <Button
        style={{
          marginBottom: '30px',
          border: '1px solid #BB1333',
          color: '#BB1333',
        }}
        onClick={() => history.goBack()}
      >
        <Icon type="left" />
        Back
      </Button>
      {searchResults[2] ? (
        <h3 className="searchResultsP">
          Showing results for &quot;
          {searchResults[2]}
          &quot;.
        </h3>
      ) : null}

      <CardContainer>
        {!isSearching && (!searchResults[0] || !searchResults[0].length) ? (
          <div>No results found.</div>
        ) : !isSearching && searchResults[1] === 'companies' ? (
          searchResults[0].map(company => (
            <CompanySearchCard
              key={company.id}
              name={company.name}
              location={company.location}
              website={company.website}
              type={company.type}
              id={company.id}
              history={history}
            />
          ))
        ) : !isSearching && searchResults[1] === 'salaries' ? (
          searchResults[0].map(salaryReview => (
            <SalarySearchCard
              key={salaryReview.id}
              name={salaryReview.name}
              salary={salaryReview.salary}
              job_title={salaryReview.job_title}
              id={salaryReview.id}
              company_id={salaryReview.company_id}
              currency={salaryReview.currency}
              history={history}
            />
          ))
        ) : !isSearching && searchResults[1] === 'interviews' ? (
          searchResults[0].map(interviewReview => (
            <InterviewSearchCard
              key={interviewReview.id}
              name={interviewReview.name}
              text={interviewReview.text}
              website={interviewReview.website}
              type={interviewReview.type}
              id={interviewReview.id}
              interest={interviewReview.interest}
              history={history}
            />
          ))
        ) : (
          <SpinContainer>
            <Spin size="large" />
          </SpinContainer>
        )}
      </CardContainer>
    </Container>
  );
};

export default withRouter(connect(state => state)(SearchResults));
const Container = styled.div`
  height: 100%;
  width: 100%;
`;

const CardContainer = styled.div`
  display: flex;
  flex-direction: table-row;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: flex-start;
  height: 80%;
  width: 100%;

  .searchResultsP {
    font-weight: bold;
  }
`;

const SpinContainer = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;
