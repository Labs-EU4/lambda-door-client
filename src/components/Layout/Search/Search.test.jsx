import React from 'react';
import * as rtl from '@testing-library/react';
import Search from './Search';
import '@testing-library/jest-dom/extend-expect';
import { renderWithRedux } from '../../../utils/testingHelpers';

beforeEach(rtl.cleanup);

describe('Search', () => {
  it('renders without crashing', () => {
    renderWithRedux(<Search />);
  });

  it('renders correctly', () => {
    expect(renderWithRedux(<Search />).baseElement).toMatchSnapshot();
  });
});
