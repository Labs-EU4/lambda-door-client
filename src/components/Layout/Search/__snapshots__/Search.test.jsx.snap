// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa fXjWTE"
    >
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="search-category"
        >
          <div
            class="ant-select ant-select-enabled"
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
              tabindex="0"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection-selected-value"
                  style="display: block; opacity: 1;"
                  title="Companies"
                >
                  Companies
                </div>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  class="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </div>
        <div
          class="search-field"
        >
          <input
            class="ant-input"
            name="search_query"
            placeholder="Company Name, Location"
            type="text"
            value=""
          />
        </div>
        <button
          class="ant-btn"
          type="button"
        >
          <span>
            Search
          </span>
        </button>
        <div
          class="sc-bwzfXH fbZpGr"
        >
          <i
            aria-label="icon: close"
            class="anticon anticon-close"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </i>
        </div>
      </form>
    </div>
  </div>
</body>
`;
